src/function/open-referral-function/src/FamilyHubs.OpenReferral.Function/Functions/TriggerPullServicesWebhook.cs

```diff
@@ -0,0 +1,67 @@
+using System.Net;
+using FamilyHubs.OpenReferral.Function.ClientServices;
+using FamilyHubs.OpenReferral.Function.Entities;
+using FamilyHubs.OpenReferral.Function.Repository;
+using FamilyHubs.ServiceDirectory.Data.Entities.Staging;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+using Microsoft.Azure.Functions.Worker.Http;
+using Newtonsoft.Json.Linq;
+
+namespace FamilyHubs.OpenReferral.Function.Functions;
+
+public class TriggerPullServicesWebhook(
+    ILogger<TriggerPullServicesWebhook> logger,
+    IHsdaApiService hsdaApiService,
+    IFunctionDbContext functionDbContext)
+{
+    [Function("TriggerPullServicesWebhook")]
+    public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Function, "POST")] HttpRequestData req)
+    {
+        logger.LogInformation("[ApiReceiver] HTTP Trigger Function Started");
+
+        (HttpStatusCode httpStatusCode, JArray? services) = await hsdaApiService.GetServices();
+
+        if (httpStatusCode != HttpStatusCode.OK) return req.CreateResponse(httpStatusCode);
+
+        List<ServiceJson> servicesById = await hsdaApiService.GetServicesById(services!);
+
+        if (servicesById.Count == 0)
+        {
+            logger.LogInformation("Getting the services by ID returned no results!");
+            return req.CreateResponse(HttpStatusCode.NotFound);
```

Comment:
Same here RE returning 404 when no services are found. Also, since the HdsaApiService handles empty results, I don't think this block will ever be hit (it'd never go further than line 23 since the hsdaApiService handles it)

Reply:
HsdaApiService from line 23 handles the `/services` endpoint returning empty results, this one is the `/services/{id}` endpoint returning empty

Reply:
Changed to 204 NoContent: [fd4cb4a](https://github.com/DFE-Digital/fh-services/pull/63/commits/fd4cb4abe97699fe5bb464c00b5a3b852ea71787)